components/email/email-form/index.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from 'react'
+
+import API from '@lib/api/api-endpoints'
+import Button from '@components/generic/button/button'
+import DynamicFieldEdit from '@components/generic/dynamic-field/dynamic-field-edit'
+import { UserType } from '@lib/types'
+import fetch from '@lib/fetcher'
+
+type PropTypes = {
+  onSubmitted: () => void,
+  onCancel: () => void,
+  user: UserType
+}
+
+export default function EmailForm({
+  onSubmitted,
+  onCancel,
+  user
+}: PropTypes) {
+  const [from, setFrom] = useState(user ? user.email : '')
+  const [to, setTo] = useState('')
+  const [cc, setCc] = useState('')
+  const [bcc, setBcc] = useState('')
+  const [subject, setSubject] = useState('')
+  const [html, setHtml] = useState('')
+
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+  const [errorEmptyFields, setErrorEmptyFields] = useState(false)
+
+  useEffect(() => {
+    setFrom(user ? user.email : '')
+  }, [user])
+
+  const onSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
+    ev.preventDefault()
+
+    if ((!to && !cc && !bcc) || !html || !subject) {
+      setErrorEmptyFields(true)
+      return
+    } else {
+      setErrorEmptyFields(false)
+    }
+
+    setLoading(true)
+    setError(false)
+
+    fetch(API.email, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        from,
+        to,
+        cc,
+        bcc,
+        subject,
+        html
+      }),
+    })
+      .then(() => {
+        setError(false)
+        setLoading(false)
+        onSubmitted()
+      })
+      .catch(() => {
+        setError(true)
+        setLoading(false)
+      })
+  }
+
+
+  const onClickCancel = (ev: React.MouseEvent<HTMLButtonElement>) => {
+    ev.preventDefault()
+    ev.stopPropagation()
+    onCancel()
+  }
+
+  return (
+    <>
+      <div className="email-form">
+        <h3>New email</h3>
+        <form onSubmit={onSubmit}>
+        <DynamicFieldEdit
```

Comment:
maybe it's better to define config with fields and then use the loop to call DynamicFieldEdit component. like it was done [there](https://github.com/nucleo-org/edge-next/blob/master/components/user/admin-user/new/user-form.tsx)

Reply:
no its more difficult to style